docs/architecture/blazor-for-web-forms-developers/security-authentication-authorization.md

```diff
@@ -1,15 +1,379 @@
+
+An authorization policy consists of one or more requirements. Policies are registered as part of the authorization service configuration in the `ConfigureServices` method of `Startup.cs`. For example, the following code snippet configures a policy called "CanadiansOnly" which has the requirement that the user have the Country claim with the value of "Canada".
+
+```csharp
+services.AddAuthorization(options =>
+{
+    options.AddPolicy("CanadiansOnly", policy => policy.RequireClaim(ClaimTypes.Country, "Canada"));
+});
+```
+
+You can [learn more about how to create custom policies in the documentation](https://docs.microsoft.com/aspnet/core/security/authorization/policies).
+
+Whether you're using policies or roles, you can specify that a particular page in your Blazor application require that role or policy with the `[Authorize]` attribute, applied with the `@attribute` directive.
+
+Requiring a role:
+
+```csharp
+@attribute [Authorize(Roles ="administrators")]
+```
+
+Requiring a policy be satisfied:
+
+```csharp
+@attribute [Authorize(Policy ="CanadiansOnly")]
+```
+
+If you need access to a user's authentication state, roles, or claims you can access these by injecting the `AuthenticationStateProvider` into your component:
+
+```csharp
+@using Microsoft.AspNetCore.Components.Authorization
+@inject AuthenticationStateProvider AuthenticationStateProvider
+```
+
+With the provider in place, you can gain access to the user with the following code:
+
+```csharp
+AuthenticationState authState = await AuthenticationStateProvider.GetAuthenticationStateAsync();
+ClaimsPrincipal user = authState.User;
+
+if (user.Identity.IsAuthenticated)
+{
+  // work with user.Claims and/or user.Roles
+}
+```
+
+To work with users and claims you may also need to inject a `UserManager<T>` (use `IdentityUser` for default) which you can use to enumerate and modify claims for a user. First inject the type and assign it to a property:
+
+```csharp
+@inject UserManager<IdentityUser> MyUserManager
+```
+
+Then use it to work with the user's claims. The following sample shows how to add and persist a claim on a user:
+
+```csharp
+private async Task AddCountryClaim()
+{
+    var authState = await AuthenticationStateProvider.GetAuthenticationStateAsync();
+    var user = authState.User;
+    var identityUser = await MyUserManager.FindByNameAsync(user.Identity.Name);
+
+    if (!user.HasClaim(c => c.Type == ClaimTypes.Country))
+    {
+        // stores the claim in the cookie
+        ClaimsIdentity id = new ClaimsIdentity();
+        id.AddClaim(new Claim(ClaimTypes.Country, "Canada"));
+        user.AddIdentity(id);
+
+        // save the claim in the database
+        await MyUserManager.AddClaimAsync(identityUser, new Claim(ClaimTypes.Country, "Canada"));
+    }
+}
+```
+
+If you need to work with roles, follow the same approach. You may need to inject a `RoleManager<T>` (use `IdentityRole` for default type) to list and manage the roles themselves.
```

Comment:
Again, you can only use `RoleManager<T>` from your components in a Blazor Server app, not Blazor WebAssembly.